ex2_hal/athena/equipment_handler/source/deployablecontrol.c

```diff
@@ -20,55 +20,105 @@
         gioSetBit(hetPORT1, 8, 1);
+
+        // Wait
         vTaskDelay(DEPLOYABLE_BURNWIRE_DELAY_MS);
+
+        // Read the current through the burnwire
+        eps_refresh_instantaneous_telemetry();
+        eps_instantaneous_telemetry_t eps = get_eps_instantaneous_telemetry();
+
+        // Deactivate burnwire
         gioSetBit(hetPORT1, 8, 0);
+
+        // Store the current through the burnwire
+        *burnwire_current = eps.curOutput[8];
         break;
-    case UHF_P:
+    }
+    case UHF_P:{
         gioSetBit(hetPORT2, 5, 1);
         vTaskDelay(DEPLOYABLE_BURNWIRE_DELAY_MS);
+        eps_refresh_instantaneous_telemetry();
+        eps_instantaneous_telemetry_t eps = get_eps_instantaneous_telemetry();
         gioSetBit(hetPORT2, 5, 0);
+
+        *burnwire_current = eps.curOutput[8];
         break;
-    case UHF_Z:
+    }
+    case UHF_Z:{
         gioSetBit(hetPORT1, 26, 1);
         vTaskDelay(DEPLOYABLE_BURNWIRE_DELAY_MS);
+        eps_refresh_instantaneous_telemetry();
+        eps_instantaneous_telemetry_t eps = get_eps_instantaneous_telemetry();
         gioSetBit(hetPORT1, 26, 0);
+
+        *burnwire_current = eps.curOutput[8];
         break;
-    case Payload:
+    }
+    case Payload:{
         gioSetBit(hetPORT1, 14, 1);
         vTaskDelay(DEPLOYABLE_BURNWIRE_DELAY_MS);
+        eps_refresh_instantaneous_telemetry();
+        eps_instantaneous_telemetry_t eps = get_eps_instantaneous_telemetry();
         gioSetBit(hetPORT1, 14, 0);
+
+        *burnwire_current = eps.curOutput[8];
         break;
-    case UHF_S:
+    }
+    case UHF_S:{
         gioSetBit(gioPORTA, 0, 1);
         vTaskDelay(DEPLOYABLE_BURNWIRE_DELAY_MS);
+        eps_refresh_instantaneous_telemetry();
+        eps_instantaneous_telemetry_t eps = get_eps_instantaneous_telemetry();
         gioSetBit(gioPORTA, 0, 0);
+
+        *burnwire_current = eps.curOutput[8];
         break;
-    case UHF_N:
+    }
+    case UHF_N:{
         gioSetBit(hetPORT1, 22, 1);
         vTaskDelay(DEPLOYABLE_BURNWIRE_DELAY_MS);
+        eps_refresh_instantaneous_telemetry();
+        eps_instantaneous_telemetry_t eps = get_eps_instantaneous_telemetry();
         gioSetBit(hetPORT1, 22, 0);
+
+        *burnwire_current = eps.curOutput[8];
         break;
-    case Starboard:
+    }
+    case Starboard:{
         gioSetBit(hetPORT2, 1, 1);
         vTaskDelay(DEPLOYABLE_BURNWIRE_DELAY_MS);
+        eps_refresh_instantaneous_telemetry();
+        eps_instantaneous_telemetry_t eps = get_eps_instantaneous_telemetry();
         gioSetBit(hetPORT2, 1, 0);
+
+        *burnwire_current = eps.curOutput[8];
         break;
-    case DFGM:
+    }
+    case DFGM:{
         gioSetBit(hetPORT1, 16, 1);
         vTaskDelay(DEPLOYABLE_BURNWIRE_DELAY_MS);
+        eps_refresh_instantaneous_telemetry();
+        eps_instantaneous_telemetry_t eps = get_eps_instantaneous_telemetry();
         gioSetBit(hetPORT1, 16, 0);
+
+        *burnwire_current = eps.curOutput[8];
```

Comment:
is the current always measured from curOutput[8]? And then the current gets routed to different hetPORTs?

Reply:
Yes will add!

Reply:
And yes all of the deployables are connected to channel 9 (index 8). The different hetPORTS basically serve to open a gate and allow the current to flow across a particular resistor